smart-py/objkt_swap.py

```diff
@@ -17,135 +33,271 @@ def __init__(self, objkt, hdao, manager, metadata, curate):
-        sp.verify((params.objkt_amount > 0))
-        self.fa2_transfer(self.data.objkt, sp.sender, sp.to_address(sp.self), params.objkt_id, params.objkt_amount)
-        self.data.swaps[self.data.swap_id] = sp.record(issuer=sp.sender, objkt_id=params.objkt_id, objkt_amount=params.objkt_amount, xtz_per_objkt=params.xtz_per_objkt)
+        sp.verify(params.objkt_amount > 0)
+        # sp.verify(params.objkt_amount == 1)
+
+        self.fa2_transfer(
+            self.data.objkt,
+            sp.sender,
+            sp.to_address(sp.self),
+            params.objkt_id,
+            params.objkt_amount
+        )
+
+        self.data.swaps[self.data.swap_id] = sp.record(
+            issuer=sp.sender,
+            objkt_id=params.objkt_id,
+            objkt_amount=params.objkt_amount,
+            xtz_per_objkt=params.xtz_per_objkt
+        )
         self.data.swap_id += 1
-    
+
     @sp.entry_point
     def cancel_swap(self, params):
-        sp.verify( (self.data.swaps[params].issuer == sp.sender) )
-        self.fa2_transfer(self.data.objkt, sp.to_address(sp.self), sp.sender, self.data.swaps[params].objkt_id, self.data.swaps[params].objkt_amount)
-        
+        sp.verify(self.data.swaps[params].issuer == sp.sender)
+
+        self.fa2_transfer(
+            self.data.objkt,
+            sp.to_address(sp.self),
+            sp.sender,
+            self.data.swaps[params].objkt_id,
+            self.data.swaps[params].objkt_amount
+        )
+
         del self.data.swaps[params]
-        
+
     @sp.entry_point
     def collect(self, params):
-        sp.verify( (params.objkt_amount > 0) & (sp.sender != self.data.swaps[params.swap_id].issuer) )
-        
+        sp.verify(
+            (params.objkt_amount > 0) &
+            # (params.objkt_amount == 1) &
+            (sp.sender != self.data.swaps[params.swap_id].issuer)
+        )
+
         sp.if (self.data.swaps[params.swap_id].xtz_per_objkt != sp.tez(0)):
-        
-            self.objkt_amount = sp.fst(sp.ediv(sp.amount, self.data.swaps[params.swap_id].xtz_per_objkt).open_some())
-            
-            self.amount = self.objkt_amount * sp.fst(sp.ediv(self.data.swaps[params.swap_id].xtz_per_objkt, sp.mutez(1)).open_some())
-            
-            sp.verify((params.objkt_amount == self.objkt_amount) & (sp.amount == sp.utils.nat_to_mutez(self.amount)) & (sp.amount > sp.tez(0)))
+
+            self.objkt_amount = sp.fst(
+                sp.ediv(
+                    sp.amount,
+                    self.data.swaps[params.swap_id].xtz_per_objkt
+                ).open_some()
+            )
+
+            self.amount = self.objkt_amount * sp.fst(
+                sp.ediv(
+                    self.data.swaps[params.swap_id].xtz_per_objkt,
+                    sp.mutez(1)
+                ).open_some()
+            )
+
+            sp.verify(
+                (params.objkt_amount == self.objkt_amount) &
+                (sp.amount == sp.utils.nat_to_mutez(self.amount)) &
```

Comment:
This is not about formatting, but as soon as you asked for review and this is important, I decided to comment about collect logic.
There should be check for `params.objkt_amount <= self.data.swaps[params.swap_id].objkt_amount`